manifests/repository.pp

```diff
@@ -1,14 +1,41 @@
 # == Class: bareos::repository
-# This class manages the bareos repository
-# Parameters should be configured in the upper class `::bareos`.
 #
-# This class will be automatically included when a resource is defined.
-# It is not intended to be used directly by external resources like node definitions or other modules.
+# @summary
+#   This class manages the bareos repository
+#   Parameters should be configured in the upper class `::bareos`.
+#
+#   This class will be automatically included when a resource is defined.
+#   It is not intended to be used directly by external resources like node definitions or other modules.
+#
+# @param release
+#   The major bareos release version which should be used
+# @param gpg_key_fingerprint
+#   The GPG fingerprint of the repos key
+# @param subscription
+#   Activate the (paid) subscription repo. Otherwise the opensource repos will be selected
+# @param username
+#   The major bareos release version which should be used
+# @param password
+#   The major bareos release version which should be used
+#
 class bareos::repository (
-  $release = 'latest',
-  $gpg_key_fingerprint = undef,
+  String              $release             = 'latest',
+  Optional[String[1]] $gpg_key_fingerprint = undef,
+  Boolean             $subscription        = false,
+  Optional[String]    $username            = undef,
+  Optional[String]    $password            = undef,
 ) {
-  $url = "http://download.bareos.org/bareos/release/${release}/"
+  $scheme = 'http://'
+  if $subscription {
+    if $username and $password {
```

Comment:
I think we can shorten this if/else for the username/password if we reverse the logic:
```puppet
if empty($username) or empty($password) {
  fail('For Bareos subscription repos both username and password are required.')
}
```

Reply:
done